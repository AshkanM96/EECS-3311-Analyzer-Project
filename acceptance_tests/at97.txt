sum
open_set_enumeration
multiply
integer_constant(2)
add
integer_constant(3)
integer_constant(4)
conjoin
disjoin
boolean_constant(true)
boolean_constant(false)
boolean_constant(true)
close_set_enumeration
type_check
evaluate
reset
sum
open_set_enumeration
multiply
integer_constant(2)
add
integer_constant(3)
integer_constant(4)
multiply
integer_constant(-5)
subtract
negation
integer_constant(4)
integer_constant(8)
close_set_enumeration
type_check
evaluate
