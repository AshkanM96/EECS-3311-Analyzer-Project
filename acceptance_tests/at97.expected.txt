  Expression currently specified: ?
  Report: Expression is initialized.
->sum
  Expression currently specified: (+ ?)
  Report: OK.
->open_set_enumeration
  Expression currently specified: (+ {?})
  Report: OK.
->multiply
  Expression currently specified: (+ {(? * nil)})
  Report: OK.
->integer_constant(2)
  Expression currently specified: (+ {(2 * ?)})
  Report: OK.
->add
  Expression currently specified: (+ {(2 * (? + nil))})
  Report: OK.
->integer_constant(3)
  Expression currently specified: (+ {(2 * (3 + ?))})
  Report: OK.
->integer_constant(4)
  Expression currently specified: (+ {(2 * (3 + 4)), ?})
  Report: OK.
->conjoin
  Expression currently specified: (+ {(2 * (3 + 4)), (? && nil)})
  Report: OK.
->disjoin
  Expression currently specified: (+ {(2 * (3 + 4)), ((? || nil) && nil)})
  Report: OK.
->boolean_constant(True)
  Expression currently specified: (+ {(2 * (3 + 4)), ((True || ?) && nil)})
  Report: OK.
->boolean_constant(False)
  Expression currently specified: (+ {(2 * (3 + 4)), ((True || False) && ?)})
  Report: OK.
->boolean_constant(True)
  Expression currently specified: (+ {(2 * (3 + 4)), ((True || False) && True), ?})
  Report: OK.
->close_set_enumeration
  Expression currently specified: (+ {(2 * (3 + 4)), ((True || False) && True)})
  Report: OK.
->type_check
  Expression currently specified: (+ {(2 * (3 + 4)), ((True || False) && True)})
  Report: (+ {(2 * (3 + 4)), ((True || False) && True)}) is not type-correct.
->evaluate
  Expression currently specified: (+ {(2 * (3 + 4)), ((True || False) && True)})
  Report: Error (Expression is not type-correct).
->reset
  Expression currently specified: ?
  Report: OK.
->sum
  Expression currently specified: (+ ?)
  Report: OK.
->open_set_enumeration
  Expression currently specified: (+ {?})
  Report: OK.
->multiply
  Expression currently specified: (+ {(? * nil)})
  Report: OK.
->integer_constant(2)
  Expression currently specified: (+ {(2 * ?)})
  Report: OK.
->add
  Expression currently specified: (+ {(2 * (? + nil))})
  Report: OK.
->integer_constant(3)
  Expression currently specified: (+ {(2 * (3 + ?))})
  Report: OK.
->integer_constant(4)
  Expression currently specified: (+ {(2 * (3 + 4)), ?})
  Report: OK.
->multiply
  Expression currently specified: (+ {(2 * (3 + 4)), (? * nil)})
  Report: OK.
->integer_constant(-5)
  Expression currently specified: (+ {(2 * (3 + 4)), (-5 * ?)})
  Report: OK.
->subtract
  Expression currently specified: (+ {(2 * (3 + 4)), (-5 * (? - nil))})
  Report: OK.
->negation
  Expression currently specified: (+ {(2 * (3 + 4)), (-5 * ((! ?) - nil))})
  Report: OK.
->integer_constant(4)
  Expression currently specified: (+ {(2 * (3 + 4)), (-5 * ((! 4) - ?))})
  Report: OK.
->integer_constant(8)
  Expression currently specified: (+ {(2 * (3 + 4)), (-5 * ((! 4) - 8)), ?})
  Report: OK.
->close_set_enumeration
  Expression currently specified: (+ {(2 * (3 + 4)), (-5 * ((! 4) - 8))})
  Report: OK.
->type_check
  Expression currently specified: (+ {(2 * (3 + 4)), (-5 * ((! 4) - 8))})
  Report: (+ {(2 * (3 + 4)), (-5 * ((! 4) - 8))}) is not type-correct.
->evaluate
  Expression currently specified: (+ {(2 * (3 + 4)), (-5 * ((! 4) - 8))})
  Report: Error (Expression is not type-correct).
